core/test_network/src/lib.rs

```diff
@@ -422,24 +441,26 @@ where
         }
     }
 
-    /// Starts peer with config, permissions and temporary directory
-    ///
-    /// # Panics
-    /// - Starting [`Iroha`] instance fails.
-    /// - Block store path not readable
-    /// - [`Iroha::start_as_task`] failed or produced empty job handle.
-    /// - `receiver` fails to produce a message.
-    ///
-    /// # TODO
-    /// Use *Builder* pattern (#2238)
-    pub async fn start_with_config_permissions_dir(
-        &mut self,
-        configuration: Configuration,
-        genesis: Option<G>,
-        instruction_validator: impl Into<IsInstructionAllowedBoxed<W>> + Send + 'static,
-        query_validator: impl Into<IsQueryAllowedBoxed<W>> + Send + 'static,
-        temp_dir: Arc<TempDir>,
-    ) {
+    /// Starts a peer with arguments.
+    async fn start(&mut self, arguments: PeerArgs<W, G>) {
```

Comment:
What about to implement `Option` unwrapping not inside `Peer::start()` but inside `PeerBuilder`, as we discussed on the call? I think, that will be more logical

Reply:
I forgot about it. Added.